app/lib/synthetic_branding_group_populator.rb

```diff
@@ -0,0 +1,107 @@
+##
+# SyntheticBrandingGroupPopulator: class to populate branding groups using the output of SyntheticStudyPopulator
+##
+
+class SyntheticBrandingGroupPopulator
+  BASE_BRANDING_GROUPS_CONFIG_PATH = Rails.root.join('db', 'seed', 'synthetic_branding_groups')
+
+  # populate all branding groups using synthetic studies
+  #
+  # @param user (User)                 => User account to own the branding groups
+  # @param overwrite_groups (Boolean)  => T/F for destroying/recreating any existing groups
+  # @param overwrite_studies (Boolean) => T/F for destroying/recreating any existing studies
+  def self.populate_all(user: User.first, overwrite_groups: true, overwrite_studies: false)
+    # populate all studies first to ensure they exist
+    SyntheticStudyPopulator.populate_all(overwrite: overwrite_studies)
+    synthetic_studies_ids = self.collect_synthetic_studies.pluck(:id)
+    # divide into two groups, one for human data, another for mouse
+    human_studies = Study.where(name: /human/i, :id.in => synthetic_studies_ids)
+    mouse_studies = Study.where(name: /(mouse|murine)/i, :id.in => synthetic_studies_ids)
+    if human_studies.empty? || mouse_studies.empty?
+      raise ArgumentError.new("Required synthetic studies not present, please manually run SyntheticStudyPopulator.populate_all")
+    end
+    group_dirs = Dir.glob(BASE_BRANDING_GROUPS_CONFIG_PATH.join('*')).select {|f| File.directory? f}
+    group_dirs.each do |group_dir|
+      # detect if this is the human or mouse group
+      group_type = group_dir.split('/').last.split('_').first
+      study_list = group_type == 'human' ? human_studies : mouse_studies
+      self.populate(group_dir, user: user, study_list: study_list, overwrite: overwrite_groups)
+    end
+  end
+
+  # remove all synthetic branding groups, leaving studies in place
+  def self.remove_all
+    config_dirs = Dir.glob(BASE_BRANDING_GROUPS_CONFIG_PATH.join('*')).select {|f| File.directory? f}
+    config_dirs.each do |config_dir|
```

Comment:
Lets factor out a common method "get_all_branding_group_configs", that handles going through the directories, getting the files, and parsing the configuration files

Reply:
568c460, 7290786